File: JPA/springJPA_webService/src/main/java/me/kzv/springjpa_webservice/controller/SampleController.java
package me.kzv.springjpa_webservice.controller;

import lombok.extern.log4j.Log4j2;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;


@Log4j2
@RequestMapping("/sample")
@Controller
public class SampleController {

    //test
    @GetMapping
    public void ex(){
        log.info("ex............");
    }
}

File: kopring/kopring-practice/src/main/kotlin/me/kzv/kopringpractice/controller/SampleController.kt
package me.kzv.kopringpractice.controller

import jakarta.servlet.http.HttpServletResponse
import org.slf4j.LoggerFactory
import org.springframework.http.HttpHeaders
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.*

@Controller
class SampleController {
    val logger = LoggerFactory.getLogger(this.javaClass)

    @GetMapping("/sample/re")
    fun redirectSample(response: HttpServletResponse) {
        logger.info("redirectSample")
        response.addHeader("X-TOKEN", "asdfjlkaskld012")
        response.sendRedirect("/sample")
    }

    @GetMapping("/sample")
    fun sampleSample(@RequestHeader header: HttpHeaders, model: Model): String {
        logger.info("sample ${header["X-TOKEN"]}")
        model.addAttribute("token", header["X-TOKEN"])
        return "sample"
    }

    @GetMapping("/")
    fun index(model: Model): String {
        return "index"
    }

    @PostMapping("/sample")
    fun postSample(@RequestParam("token") token: String, model: Model): String {
        logger.info("rediredted with: $token")
        model.addAttribute("token", token)
        return "sample"
    }
}